functional-tests/src/test/groovy/pages/modules/HeaderModule.groovy

```diff
@@ -38,47 +38,26 @@ class HeaderModule extends Module {
    *   to wait for some period of time before assuming the notification count is intentionally absent.  Otherwise
    *   you may get false-positives.  To improve performance, the wait time should be as little as safely possible.
    *
-   * @param int wait how long to wait for the notification count icon to load, before assuming it is not going to.
-   *   (optional, default: 2)
+   * @param int wait how long to wait for the async notification request to finish, and the icon to render, before
+   *   assuming it is not going to (if there are no notifications). (optional, default: 3)
    * @return notification count integer
    * @throws NumberFormatException if the parsed notification count string fails to be cast to an Integer.
    */
-  Integer getNotificationCount(int wait=2) {
+  Integer getNotificationCount(int wait=3) {
     Integer count = 0
 
     try {
-      count = waitFor(wait) {
+      // We cant use 'waitFor(3)' in this case because the notification icon may exist, but it may not have been
+      // updated yet by the async notification request which has not yet finished.  So we need to manually enforce a
+      // wait using 'sleep(3000)'
+      sleep(wait * 1000)
+      count = waitFor(0) { // convenient way to cause a predictable exception if this selector never exists.
```

Comment:
This function is the main change. Basically moving from a wait based on the appearance of the notification count icon, to a fixed wait no matter what, to allow async notification requests to finish before reading the notification count.